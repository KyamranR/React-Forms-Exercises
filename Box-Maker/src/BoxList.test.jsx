import { render, fireEvent } from "@testing-library/react";
import BoxList from "./BoxList";

test("renders without crashing", () => {
  render(<BoxList />);
});

test("matches snapshot", () => {
  const { asFragment } = render(<BoxList />);
  expect(asFragment()).toMatchSnapshot();
});

test("can add a new box", () => {
  const { getByLabelText, getByText, queryByText } = render(<BoxList />);

  const widthInput = getByLabelText("Width:");
  const heightInput = getByLabelText("Height:");
  const colorInput = getByLabelText("Background Color:");
  const addButton = getByText("Add Box");

  fireEvent.change(widthInput, { target: { value: "100" } });
  fireEvent.change(heightInput, { target: { value: "100" } });
  fireEvent.change(colorInput, { target: { value: "#0000FF" } });
  fireEvent.click(addButton);

  expect(queryByText("X")).toBeInTheDocument();
});

test("can remove a box", () => {
  const { getByLabelText, getByText, queryByText } = render(<BoxList />);

  const widthInput = getByLabelText("Width:");
  const heightInput = getByLabelText("Height:");
  const colorInput = getByLabelText("Background Color:");
  const addButton = getByText("Add Box");

  fireEvent.change(widthInput, { target: { value: "100" } });
  fireEvent.change(heightInput, { target: { value: "100" } });
  fireEvent.change(colorInput, { target: { value: "#0000FF" } });
  fireEvent.click(addButton);

  const removeButton = getByText("X");
  expect(removeButton).toBeInTheDocument();

  fireEvent.click(removeButton);
  expect(queryByText("X")).not.toBeInTheDocument();
});
