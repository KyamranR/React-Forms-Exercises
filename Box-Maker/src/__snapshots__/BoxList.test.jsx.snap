// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <form>
      <label
        for="width"
      >
        Width:
      </label>
      <input
        id="width"
        name="width"
        type="text"
        value=""
      />
      <label
        for="height"
      >
        Height:
      </label>
      <input
        id="height"
        name="height"
        type="text"
        value=""
      />
      <label
        for="backgroundColor"
      >
        Background Color:
      </label>
      <input
        id="backgroundColor"
        name="backgroundColor"
        type="color"
        value="#000000"
      />
      <button>
        Add Box
      </button>
    </form>
    <div />
  </div>
</DocumentFragment>
`;
